src/common/imageio.c

```diff
@@ -1108,7 +1108,19 @@ int dt_imageio_export_with_flags(const int32_t imgid, const char *filename,
   format_params->width = processed_width;
   format_params->height = processed_height;
 
-  if(!ignore_exif)
+  // Check if all the metadata export flags are set for AVIF/EXR/JPEG XL (opt-in)
+  // TODO: this is a workround as these formats do not support fine grained metadata control through
+  // dt_exif_xmp_attach_export() below due to lack of exiv2 write support
+  gboolean md_flags_set = TRUE;
+  if(!strcmp(format->mime(NULL), "image/avif") || !strcmp(format->mime(NULL), "image/x-exr")
+     || !strcmp(format->mime(NULL), "image/jxl"))
+  {
+    md_flags_set = metadata->flags
```

Comment:
I would rather suggest to add DT_META_ALL (or whatever) to dt_metadata_id enum definition with corresponding comment, because if new flag will be added in the future, this line in imageio.c won't be updated for sure.

Reply:
Let's first decide on the right logic. As I said, there is a question mark about "omit hierarchy", in which case there will be some XOR-ing going on, and `DT_META_ALL` will not make total sense then perhaps?

Reply:
Yeah, I don't know what to suggest about "omit hierarchy". But if we decide on some special handling of that flag, I still suggest to add DT_META_ALL to dt_metadata_id, and do that special handling with bit operations here in imageio.c. But sure, let's first agree on logic.